src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -126,10 +126,7 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         }
 
         // now, remove the commands with the IDs we found
-        for (const auto id : IdsToRemove)
-        {
-            _ActionMap.erase(id);
-        }
+        std::erase_if(_ActionMap, [&IdsToRemove](const auto& pair) { return IdsToRemove.contains(pair.first); });

```

Comment:
That's not exactly what I meant with my previous suggestion. The loop above this one goes through each `_ActionMap` entry to decide whether to add it to the `IdsToRemove` list. The `userID` it adds is always the key value of the current iteration.

In other words, this works too:
```cpp
std::erase_if(_ActionMap, [&](const auto& [userID, userCmd]) {
    const auto userCmdImpl{ get_self<Command>(userCmd) };
    // Note we don't need to explicitly check for the origin tag here since we only generate IDs for user actions,
    // ...
        // ...
            // ...
            // add this ID to our set of IDs to remove
            return true;
        }
    }
    return false;
});
```